Source/WebCore/platform/mac/ThemeMac.mm

```diff
@@ -652,11 +662,15 @@ static LengthSize switchSize(const LengthSize& zoomedSize, float zoomFactor)
         break;
     }
     case StyleAppearance::Switch: {
-        NSControlSize controlSize = controlSizeFromPixelSize(switchSizes(), zoomRectSize, zoomFactor);
-        IntSize zoomedSize = switchSizes()[controlSize];
+        auto isVertical = states.contains(ControlStyle::State::VerticalWritingMode);
+        auto logicalZoomRectSize = isVertical ? zoomRectSize.transposedSize() : zoomRectSize;
+        auto controlSize = controlSizeFromPixelSize(switchSizes(), logicalZoomRectSize, zoomFactor);
+        auto zoomedSize = switchSizes()[controlSize];
         zoomedSize.setHeight(zoomedSize.height() * zoomFactor);
         zoomedSize.setWidth(zoomedSize.width() * zoomFactor);
-        zoomedRect = inflateRect(zoomedRect, zoomedSize, switchMargins(controlSize), zoomFactor);
+        if (isVertical)
+            zoomedSize = zoomedSize.transposedSize();
+        zoomedRect = inflateRect(zoomedRect, zoomedSize, logicalSwitchMargins(controlSize, isVertical), zoomFactor);
         break;
```

Comment:
There is something wrong with this logic as not everything is redrawn for a 10x10 vertical switch. In particular there's a bar on the right side that remains static.

(This code might also need further tidying in general, but I wanted to put it up to maybe get another review cycle before the weekend.)

Reply:
As far as I can tell this is doing the correct thing. Something with repaint seems amiss.

Reply:
Minimal test:
`data:text/html,<input%20type=checkbox%20switch%20style=width:10px;height:10px;writing-mode:vertical-rl>`

Reply:
I filed https://bugs.webkit.org/show_bug.cgi?id=266532 on this. I'm quite positive now this code is correct. It produces equivalent results for horizontal and vertical. It just goes amiss with vertical due to the repaint code.

Reply:
(The one thing I have now revised here is renamed `logicalSwitchMargins` to `visualSwitchMargins`. This is still correct otherwise.)

Reply:
Talked about this offline, none of the paint inflation code for controls accounts for flipped blocks writing mode, so we will fix separately.